base/acme/src/main/java/org/dogtagpki/acme/server/ACMEEngine.java

```diff
@@ -114,8 +110,11 @@ public static ACMEEngine getInstance() {
         return INSTANCE;
     }
 
+    public static void setInstance(ACMEEngine instance) {
```

Comment:
The INSTANCE should only be accessed and not modified

Reply:
@edewata FYI the way I am dealing with this in the EST subsystem is to maintain a `Map<String, ESTEngine>` keyed by contextPath.  `contextInitialized` and `contextDestroyed` add and remove the corresponding instance from the map.  In this way, you can have >1 instance of the subsystem running in Tomcat one stepping on the other.

Note: I just released we may need to key by the combination of hostname and path.  But you get the idea.

This PR continues the current approach.  So IMO it need not be addressed in the PR but could be a subsequent step.

Reply:
I updated my WIP branch so that we include virtual server name ***and*** context path in the map key.  I tagged the commit (so we don't lose it in a future push): https://github.com/frasertweedale/pki/commit/1918dfcc4db1879f879f63fdffbd34750588e08f

Sorry about the `HostAndContextPath` inner class but Java doesn't have tuples and a generic pair with accompanying `Comparable` method or accompanying `Comparator` implementation was overkill.